torch/csrc/jit/passes/onnx.cpp

```diff
@@ -326,10 +325,17 @@ void NodeToONNX(
           ONNXShapeTypeInference(const_node, empty_params_dict, opset_version);
           env[old] = const_node->output();
         } else {
-          // ConstantValueMap has been set in shape inference,
-          // set_constant_value_map = false here to avoid redundancy.
+          // An update in ConstantValueMap is also needed here, since
+          // the user setType can be only accessed in this step, and it
+          // should be reliable.
           MergeInferredTypeAndSetMap(
-              outputs[i], old->type(), outputs[i]->type(), false);
```

Comment:
Why is this function (`MergeInferredTypeAndSetMap`) required?

Reply:
The reason #62856 brought `MergeInferredTypeAndSetMap` here is that we can have tracer shape as early as right after symbolic_fn processing. That is, if we didn't have `MergeInferredTypeAndSetMap`, the graph here would contain nodes without any type like the following:
```
%1 : Tensor = com.microsoft::Inverse(%0), scope: __main__.CustomInverse:: 
```